### PR TITLE
Saleor App Store docs

### PR DESCRIPTION
**Pull request by peelar**

### CODE REVIEW

File: docs/developer/app-store/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Saleor App Store is a [Saleor Cloud](cloud.mdx) feature that provides an instantly installable collection of [apps](developer/extending/apps/key-concepts.mdx). Each app brings new functionality to Saleor, often by integrating with third-party services.
+
+You can find Saleor App Store under the _Apps_ tab in the Dashboard menu. [Here](https://demo.saleor.io/dashboard/apps/) is Saleor App Store for the demo instance.
+
+## Usage
+
+Saleor App Store consists of two sections:
+
+- Installed Apps - a list of apps that are currently installed on your instance
+- All Apps - a list of all apps available in the App Store
```
Comment (lkostrowski) [inline docs/developer/app-store/overview.mdx:20]: That is not true. Technically *all* Saleors (not only cloud) have "installed apps" and only Cloud has access to "All apps" section below, which is actual appstore

So we need to either remove that Appstore is "cloud only" or remove mention of installed apps

I'd follow the latter approach and rephrase it (something like "appstore is a part of Dashboard->apps page)

File: docs/developer/app-store/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Clicking the _Install_ button on any app card will trigger the installation process during which you will be asked to confirm the required app permissions. After doing so, the app will appear in the _Installed Apps_ section.
+
+If you want to deactivate or delete an app, click on the _Details_ button next to the app name in the _Installed Apps_ section. You can manage your app from there.
```
Comment (lkostrowski) [inline docs/developer/app-store/overview.mdx:24]: Its _Settings_ , not _Details_

File: docs/developer/app-store/apps/index.mdx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import CardGrid from "@site/components/CardGrid";
+
+# Apps
+
+## List
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/index.mdx:5]: Maybe something like "A Built-in apps catalogue". "List" is not a good heading

File: docs/developer/app-store/apps/index.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import CardGrid from "@site/components/CardGrid";
+
+# Apps
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/index.mdx:3]: I'd rephrase to `Appstore Apps` or something like this.

Keep in mind that its SEO-indexed and people will see this as a first page, not after clicking "app-store" section in the menu. 

Apps have several concepts in Saleor:
1. Webhooks / simple local apps
2. 3rd party / custom apps & their development
3. Using Appstore & apps provided by Saleor
4. Forking/working with Saleor Apps

All of them are separate topics, connected, but often exclusive

File: docs/developer/app-store/apps/taxes.mdx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+- [TaxJar](https://www.taxjar.com/)
+- [Avatax](https://www.avalara.com/us/en/products/calculations.html)
+- [Avatax Excise](https://developer.avalara.com/excise/) [_coming soon_]
+
+Taxes App has two views: **Channels** and **Providers**.
+
+**Channels** allows you to fill in the "ship from" address and assign a tax provider instance to the channel.
+
+**Providers** allows you to manage the tax provider instances. You can have multiple instances of the same provider.
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:20]: I'm not sure if this is a good approach, to tightly couple how app looks to the docs, it will make docs outdated soon.

I don't think its important to have "views" here. More important is what are the features, what can be done, what are assumptions and limitations. You can visit other platforms appstores for reference.

I suggest to mention:
- App allows to create more than one configuration for each provider (you can test with Dev and Prod instances of Taxjar or Avalara if you like)
- App allows you to select specific provider per channel. Its a great opportunity to mix providers for specific markets
- Visit app built-in instructions for usage details 


I think its far more accurate in readme, so people know what to expect.
App itself contains instructions and views/tabs are self explaining - we are building a product for developers

File: docs/developer/app-store/apps/taxes.mdx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+:::caution
+
+To successfully configure the Taxes App, you must have an account with one of the supported tax providers.
+
+:::
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:26]: Maybe we can have a "prerequisites" section for every app/readme?

File: docs/developer/app-store/apps/taxes.mdx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+To run the Taxes App locally:
+
+1. Follow the [_Setup_ section in the _Development_ article](developer/app-store/development.mdx#setup).
+2. Go to the app directory.
+3. Copy the `.env.example` file to `.env`.The `.env` should contain the following variables:
+
+`SECRET_KEY` (**required**)
+
+A randomly generated key for the encryption of [Settings Manager](https://github.com/saleor/saleor-app-sdk/blob/main/docs/settings-manager.md).
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:38]: Add that it can be skipped on the development but its encouraged to set it. We will add more docs when we migrate SDK here

File: docs/developer/app-store/apps/taxes.mdx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+`APL` (_optional_)
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:40]: please link docs from SDK https://github.com/saleor/saleor-app-sdk/tree/main/docs

File: docs/developer/app-store/apps/taxes.mdx (hunk 7 above)
Comment (lkostrowski) [reply docs/developer/app-store/apps/taxes.mdx:38]: also link sdk docs
https://github.com/saleor/saleor-app-sdk/blob/main/docs/settings-manager.md

File: docs/developer/app-store/apps/taxes.mdx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+Name of the chosen implementation of the [Authentication Persistence Layer](https://github.com/saleor/saleor-app-sdk/blob/main/docs/apl.md).
+
+When no value is provided, `FileAPL` is used by default. See `saleor-app.ts` in the app directory to see supported APLs.
+
+`APP_DEBUG` (_optional_)
+
+Logging level based on which the app will decide on what messages to log.
+
+The possible values are: `fatal`, `error`, `warn`, `info`, `debug`, `trace`. Read more in [pino documentation](https://getpino.io/#/docs/api?id=loggerlevel-string-gettersetter).
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:50]: by default its "silent" and no logs will be printed

File: docs/developer/app-store/apps/taxes.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+`NODE_ENV` (_optional_)
+
+Controls the environment in which the app is running.
+
+The possible values are: `development`, `production`.
+
+`PORT` (_optional_)
+
+The port on which the app will run. `3000` by default.
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:60]: I'd not mentions them, but mention that app is using next js and link to docs of next for others

https://nextjs.org/docs/basic-features/environment-variables

File: docs/developer/app-store/apps/taxes.mdx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+`ALLOWED_DOMAIN_PATTERN` (_optional_)
+
+A regex pattern that prohibits the app from being installed on a Saleor instance that does not match the pattern. If not set, all installations will be allowed.
+
+## Configuration
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:66]: As I wrote above, I think app should have good built in instructions, and here we can have brief, maintainable summary. 

Also, "Configuration" should be above "Development" sections

File: docs/developer/app-store/apps/taxes.mdx [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+#### Avatax
+
+- Instance name - a name for the instance. It will be used to identify the instance across all the views.
+- Sandbox - a flag that indicates whether the instance should use [the sandbox environment or the production environment](https://developer.avalara.com/erp-integration-guide/sales-tax-badge/authentication-in-avatax/sandbox-vs-production/).
+- Autocommit - a flag that indicates whether the instance should [automatically commit the transactions](https://developer.avalara.com/communications/dev-guide_rest_v2/commit-uncommit/).
+- Username - the username for the Avatax account. You will generate it in the Avalara dashboard (_Settings_ → _License and API keys_ → _License key_).
+- Password - the password for the Avatax account. You will generate it in the Avalara dashboard (_Settings_ → _License and API keys_ → _License key_).
+- Company code - the [company code](https://developer.avalara.com/ecommerce-integration-guide/sales-tax-badge/transactions/simple-transactions/company-codes/) for the Avatax account.
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:97]: I think this part *can* be in docs, something like "configuring providers", but I'd remove part where you explain to click a button

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: docs/developer/app-store/overview.mdx (hunk 1 above)
Comment (peelar) [reply docs/developer/app-store/overview.mdx:20]: I mean, the "All Apps" section is in the non-cloud Saleor App Store, the "install" button is just greyed out.

File: docs/developer/app-store/apps/taxes.mdx (hunk 6 above)
Comment (peelar) [reply docs/developer/app-store/apps/taxes.mdx:26]: I thought about it, and decided to handle it this way.

Would you like a "Prerequisites" section with the same content as this `caution` element, or would you like to have more information there?

File: docs/developer/app-store/apps/taxes.mdx (hunk 7 above)
Comment (peelar) [reply docs/developer/app-store/apps/taxes.mdx:38]: You mean to link the SDK docs in the `SECRET_KEY` heading? Because they are linked a line below

File: docs/developer/app-store/apps/taxes.mdx (hunk 8 above)
Comment (peelar) [reply docs/developer/app-store/apps/taxes.mdx:40]: So we want to have the docs in the titles rather than in the description (because I added it a line below), yes?

File: docs/developer/app-store/apps/taxes.mdx (hunk 7 above)
Comment (peelar) [reply docs/developer/app-store/apps/taxes.mdx:38]: Also, what happens if there is no `SECRET_KEY`? Because the SDK docs says the app will exit.

File: docs/developer/app-store/apps/taxes.mdx (hunk 6 above)
Comment (lkostrowski) [reply docs/developer/app-store/apps/taxes.mdx:22]: I think it should be standard section for every app, but content will be different, depending on apps needs

File: docs/developer/app-store/apps/taxes.mdx [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+
+# Taxes
+
+## Overview
+
+_Taxes_ is a Saleor app that allows delegating tax calculations to external tax providers. Here are the currently available providers:
+
+- [TaxJar](https://www.taxjar.com/)
+- [Avatax](https://www.avalara.com/us/en/products/calculations.html)
+- [Avatax Excise](https://developer.avalara.com/excise/) [_coming soon_]
```
Comment (lkostrowski) [inline docs/developer/app-store/apps/taxes.mdx:14]: We can add Stripe Tax

File: docs/developer/app-store/overview.mdx (hunk 1 above)
Comment (lkostrowski) [reply docs/developer/app-store/overview.mdx:20]: After merge lets forward in to someone out of the team and check if these docs (in general, not only this point) is clear

**Review by lkostrowski — APPROVED** [review]
